### PR TITLE
retry for negative TerminationGracePeriodSeconds update

### LINKED ISSUE #116126: Failure cluster [e5ba6222...] update pod with negative terminationGracePeriodSeconds should fail
##### Error text:
```
[FAILED] failed to update pod: Operation cannot be fulfilled on pods "pod-negative-grace-periodeedc00f0-25ca-4251-af98-dd9a07aa6ccb": the object has been modified; please apply your changes to the latest version and try again
In [It] at: test/e2e/node/pods.go:395 @ 02/27/23 01:24:34.539

```
#### Recent failures:
[2023/2/28 05:44:42 ci-kubernetes-e2e-gci-gce-single-flake-attempt](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-single-flake-attempt/1630323039008722944)
[2023/2/28 04:59:43 ci-kubernetes-e2e-gci-gce-containerd](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-containerd/1630311713637142528)
[2023/2/27 22:19:18 ci-kubernetes-e2e-gci-gce-kube-dns-nodecache](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-kube-dns-nodecache/1630210955210133504)
[2023/2/27 19:16:17 ci-cos-containerd-e2e-ubuntu-gce](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-cos-containerd-e2e-ubuntu-gce/1630164900850438144)
[2023/2/27 15:55:17 ci-kubernetes-e2e-gci-gce-network-proxy-http-connect](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-network-proxy-http-connect/1630114317363843072)

After https://github.com/kubernetes/kubernetes/pull/115606,

/kind failing-test

/sig node

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind failing-test

#### What this PR does / why we need it:
follow up of https://github.com/kubernetes/kubernetes/pull/115606/

#### Which issue(s) this PR fixes:

Fixes #116126

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/e2e/node/pods.go [hunk 1]
```diff
@@ -392,6 +393,14 @@ var _ = SIGDescribe("Pods Extended", func() {
...
 			_, err = podClient.PodInterface.Update(ctx, pod, metav1.UpdateOptions{})
+			if strings.Contains(err.Error(), genericregistry.OptimisticLockErrorMsg) {
+				// retry if the TerminationGracePeriodSeconds is overrided
+				// see more in https://github.com/kubernetes/kubernetes/pull/115606
+				pod, err = podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "failed to query for pod")
+				pod.Spec.TerminationGracePeriodSeconds = utilpointer.Int64(-1)
+				_, err = podClient.PodInterface.Update(ctx, pod, metav1.UpdateOptions{})
+			}
 			framework.ExpectNoError(err, "failed to update pod")
```
Comment (liggitt) [inline test/e2e/node/pods.go:404]: don't hard-code a message check, wrap the `Get + modify + Update` lines in a RetryOnConflict block

Comment (pacoxu) [reply test/e2e/node/pods.go:403]: Updated. Thanks.

File: test/e2e/node/pods.go [hunk 2]
```diff
@@ -384,14 +385,22 @@ var _ = SIGDescribe("Pods Extended", func() {
...
-			_, err = podClient.PodInterface.Update(ctx, pod, metav1.UpdateOptions{})
+			// about 26s timeout
+			updatePodBackoff := wait.Backoff{
+				Steps:    5,
+				Duration: 1 * time.Second,
+				Jitter:   2,
+			}
+			// retry if the TerminationGracePeriodSeconds is overrided
+			// see more in https://github.com/kubernetes/kubernetes/pull/115606
+			err = clientretry.RetryOnConflict(updatePodBackoff, func() error {
```
Comment (liggitt) [inline test/e2e/node/pods.go:396]: just use retry.DefaultBackoff or retry.DefaultRetry (no strong opinion which)

Comment (pacoxu) [reply test/e2e/node/pods.go]: https://github.com/kubernetes/kubernetes/blob/93a51818714630429c603e0ec74873bb243daca8/staging/src/k8s.io/client-go/util/retry/util.go#L35-L43

I chosed `retry.DefaultBackoff` as the comments matched.

**Pull request merged**